### ISSUE TITLE
Compiled.xml is missing needs a more user friendly error

### ISSUE THREAD

**Issue by nathanbunker** [immregistries/mqe#61]
When the Compiled.xml file is missing the error is very cryptic. Currently it looks like:
`Exception in thread "main" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:62)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:54)
        ... 1 more
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'databaseController' defined in URL [jar:file:/C:/mqe/dqa-message-hub-20
18.3-BETA.war!/WEB-INF/classes!/org/immregistries/dqa/hub/report/DatabaseController.class]: Instantiation of bean failed; nested exception is org.springframework.beans.Be
anInstantiationException: Failed to instantiate [org.immregistries.dqa.hub.report.DatabaseController]: Constructor threw exception; nested exception is java.lang.Exceptio
nInInitializerError
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateBean(AbstractAutowireCapableBeanFactory.java:1105)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1050)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:510)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
        at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:772)
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:839)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:538)
        at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:118)
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:766)
        at org.springframework.boot.SpringApplication.createAndRefreshContext(SpringApplication.java:361)
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:307)
        at org.immregistries.dqa.hub.DqaMessageHubApplicationConfig.main(DqaMessageHubApplicationConfig.java:20)
        ... 6 more
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.immregistries.dqa.hub.report.DatabaseController]: Constructor threw exception;
 nested exception is java.lang.ExceptionInInitializerError
        at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:163)
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:89)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateBean(AbstractAutowireCapableBeanFactory.java:1098)
        ... 21 more
Caused by: java.lang.ExceptionInInitializerError
        at org.immregistries.dqa.hub.report.DatabaseController.<init>(DatabaseController.java:37)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
        at java.lang.reflect.Constructor.newInstance(Unknown Source)
        at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:147)
        ... 23 more
Caused by: java.lang.IllegalArgumentException: You cannot build a CodeMap if the input stream is null.  Verify that you are building an input stream from a file that exis
ts.
        at org.immregistries.dqa.codebase.client.CodeMapBuilder.getDefaultCodeMap(CodeMapBuilder.java:66)
        at org.immregistries.dqa.validator.engine.codes.CodeRepository.<init>(CodeRepository.java:20)
        at org.immregistries.dqa.validator.engine.codes.CodeRepository.<clinit>(CodeRepository.java:13)`

Maybe it would be nice when this happens that the window also holds open so message is not lost.

**Comment by michellefbarber** [comment]
Possible to rename this file? Something like CompiledVaxCodes.xml? Also, is it possible to comment the XML so people know that if they modify this file locally they'll want to retain the file if they upgrade the tool? Isn't the goal to have the code update widget available and they this file will just be replaced with updated future files?

**Comment by nathanbunker** [comment]
I agree this needs to be renamed. I think the word Compiled can com out. It's probably something like VaxCodes.xml or CodeBase.xml.

**Comment by michellefbarber** [comment]
@jnarine had worked on this - ready to implement, but @nathanbunker said the implementation was causing issues with a non-MQE dependency - needs to be retooled.
